### PR TITLE
execution,plan: misc cleanups

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* move timestamp handling to plan

### CODE REVIEW

File: logicalplan/plan.go [hunk 1]
```diff
@@ -185,6 +185,23 @@ func replaceSelectors(plan parser.Expr) parser.Expr {
 			*current = &MatrixSelector{MatrixSelector: t, OriginalString: t.String()}
 		case *parser.VectorSelector:
 			*current = &VectorSelector{VectorSelector: t}
+		case *parser.Call:
```
Comment (fpetkovski) [inline logicalplan/plan.go:188]: Will this code be executed if the MergeSelectors optimizer is disabled?

Comment (MichaHoffmann) [reply logicalplan/plan.go:188]: It runs before optimizers I think

Comment (fpetkovski) [reply logicalplan/plan.go:188]: Got it, I got it confused with something else.

**Review by fpetkovski — APPROVED** [review]
Nice cleanup, keeping this sort of logic outside of execution makes complete sense.

**Comment by fpetkovski** [comment]
Tested this a bit, and I think that it introduces a bug with the remote mode since the `timestamp` function will now not be propagated.

**Comment by MichaHoffmann** [comment]
> Tested this a bit, and I think that it introduces a bug with the remote mode since the `timestamp` function will now not be propagated.

Good catch! I fixed it by rendering timestamp function in the vector selector if the flag is set!

**Review by fpetkovski — APPROVED** [review]

**Pull request merged**